### PR TITLE
Iterations pricing

### PR DESCRIPTION
**Pull request by johncalesp**
Overview of changes:

- Added a new component for the training schedule (iterations.js)  where the user can enter the number of epochs and iterations per epochs
- Modified equivalence for house in energy consumption. Depending on the user's input for the training schedule, the equivalence for house energy consumption can go from msec to years. Additionally, the reference is now a hyperlink to redirect the user to the official site.
- Changed Habitat's graph from react-vis to recharts (there were some bugs where the chart didn't show up when changing tabs). Added HorizontalBarGraph component.
- Changed Scatter plot in deployment tab from react-vis to recharts, added filters and removed training tab along with deployment button. Added ScatterGraph component.
- Moved all static data from Deployment section to providers.js
- Added more utility functions related to conversion and formatting to utils.js

**Energy consumption** 
image

**Deployment tab**
image

### CODE REVIEW

File: skyline-vscode/react-ui/src/App.js [hunk 1]
```diff
@@ -137,7 +139,7 @@ function App() {
 
     useEffect(function () {
         window.addEventListener('message', event => {
-            console.log("Message:", JSON.stringify(event.data));
+            //console.log("Message:", JSON.stringify(event.data));
```
Comment (michaelshin) [inline skyline-vscode/react-ui/src/App.js:142]: Delete instead of commenting out

File: skyline-vscode/react-ui/src/Habitat.js [hunk 2]
```diff
@@ -1,61 +1,55 @@
...
+      x: habitatData[i][1].toFixed(2),
+      fill:
+        habitatData[i][0] === "source"
+          ? colors[colors.length - 1]
+          : colors[i % (colors.length - 1)],
+    });
+  }
 
-const FlexibleXYPlot = makeWidthFlexible(XYPlot);
+  console.log(sampleData);
```
Comment (michaelshin) [inline skyline-vscode/react-ui/src/Habitat.js:33]: no need for this console.log

File: skyline-vscode/react-ui/src/Habitat.js [hunk 3]
```diff
@@ -1,61 +1,55 @@
...
+  ];
 
-import {
-    XAxis,
-    YAxis,
-    HorizontalBarSeries,
-    XYPlot,
-    makeWidthFlexible
-  } from 'react-vis';
+  let sampleData = [];
```
Comment (michaelshin) [inline skyline-vscode/react-ui/src/Habitat.js:20]: Maybe we should rename this? I was confused by the variable name and initially thought it was mock data.

File: skyline-vscode/react-ui/src/components/ScatterGraph.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+            <YAxis
+              type="number"
+              dataKey="y"
+              width={150}
+              tickFormatter={formatYAxis}
+            >
+              <Label value={ylabel} angle={-90} position="outside" />
+            </YAxis>
+            <ZAxis type="number" dataKey="z" range={[200, 300]} />
+            {/* <Scatter data={data} onClick={(e) => onClickHandler(e.payload)} /> */}
```
Comment (michaelshin) [inline skyline-vscode/react-ui/src/components/ScatterGraph.js:65]: Delete this comment

File: skyline-vscode/react-ui/src/data/mock_data.js [hunk 5]
```diff
@@ -883,94 +883,100 @@ export const profiling_data = {
...
-        total_consumption: 7.30,
-        components: [
-          { type: "ENERGY_UNSPECIFIED", consumption: 0 },
-          { type: "ENERGY_CPU_DRAM", consumption: 2.15 },
-          { type: "ENERGY_GPU", consumption: 5.15 },
-        ],
-      },
-      {
-        total_consumption: 8.20,
+      // total_consumption: 4.8,
```
Comment (michaelshin) [inline skyline-vscode/react-ui/src/data/mock_data.js:886]: Delete commented lines

File: skyline-vscode/react-ui/src/sections/EnergyConsumption.js [hunk 6]
```diff
@@ -184,13 +212,18 @@ const EnergyConsumption = ({ energyData }) => {
...
                         </Col>
                         <small>
-                          ref:
-                          https://www.epa.gov/energy/greenhouse-gas-equivalencies-calculator
+                          <a
+                            href="
+                          https://www.epa.gov/energy/greenhouse-gas-equivalencies-calculator"
+                            target="_blank"
+                          >
+                            https://www.epa.gov/energy/greenhouse-gas-equivalencies-calculator
```
Comment (michaelshin) [inline skyline-vscode/react-ui/src/sections/EnergyConsumption.js:225]: Modify it to a  line that say something like "Greenhouse gas equivalencies are based on calculations provided by [the EPA](https://www.epa.gov/energy/greenhouse-gas-equivalencies-calculator)"

File: skyline-vscode/react-ui/src/utils.js [hunk 7]
```diff
@@ -159,15 +187,38 @@ export function energy_data(currentTotal){
...
+export function calculate_training_time(numIterations, instance){
+  return numIterations * instance.x / 3.6e6 / instance.info.ngpus;
+}
+
+export function numberFormat(num){
+  const formatter = new Intl.NumberFormat("en-US", {
+    notation: "compact",
+    compactDisplay: "long",
+  });
+  // 988M
```
Comment (michaelshin) [inline skyline-vscode/react-ui/src/utils.js:212]: what is this comment for?

File: skyline-vscode/react-ui/src/utils.js [hunk 8]
```diff
@@ -159,15 +187,38 @@ export function energy_data(currentTotal){
...
     miles: `${miles.val} ${miles.scale}`,
-    household: parseFloat(Number(household).toFixed(2)),
+    household: household,
     phone: `${phone.val} ${phone.scale}`
   }
 
+}
+
+export function calculate_training_time(numIterations, instance){
+  return numIterations * instance.x / 3.6e6 / instance.info.ngpus;
```
Comment (michaelshin) [inline skyline-vscode/react-ui/src/utils.js:204]: comment what 3.6e6 is for and what units of time the output is in

File: skyline-vscode/react-ui/src/data/providers.js [hunk 9]
```diff
@@ -0,0 +1,273 @@
...
+    color: "#3498db",
+  },
+  aws: {
+    name: "Amazon Web Services",
+    logo: "resources/aws.png",
+    color: "#e67e22",
+  },
+}
+
+export const cardMemory = [
```
Comment (michaelshin) [inline skyline-vscode/react-ui/src/data/providers.js:265]: would call this gpuPropertyList

File: skyline-vscode/react-ui/src/data/providers.js [hunk 10]
```diff
@@ -0,0 +1,273 @@
...
+    id: 28,
+    provider: "aws",
+    ngpus: 4,
+    instance: "g4dn.12xlarge",
+    gpu: "t4",
+    cost: 3.912,
+  },
+];
+
+export const providers = {
```
Comment (michaelshin) [inline skyline-vscode/react-ui/src/data/providers.js:247]: Would call this cloudProviders

File: skyline-vscode/react-ui/src/data/providers.js [hunk 11]
```diff
@@ -0,0 +1,273 @@
+// GPU Instances
+// costs are 1 GPU * hr
+export const instances = [
```
Comment (michaelshin) [inline skyline-vscode/react-ui/src/data/providers.js:3]: would call this cloudInstances

**Review by michaelshin — CHANGES_REQUESTED** [review]

File: skyline-vscode/react-ui/src/App.js (hunk 1 above)
Comment (johncalesp) [reply skyline-vscode/react-ui/src/App.js]: Done, instead added other console messages for each type of event (received data, text changed, and error)

File: skyline-vscode/react-ui/src/Habitat.js (hunk 2 above)
Comment (johncalesp) [reply skyline-vscode/react-ui/src/Habitat.js]: Deleted that console.log

File: skyline-vscode/react-ui/src/Habitat.js (hunk 3 above)
Comment (johncalesp) [reply skyline-vscode/react-ui/src/Habitat.js]: Yeah, I changed to habitatProperties

File: skyline-vscode/react-ui/src/components/ScatterGraph.js (hunk 4 above)
Comment (johncalesp) [reply skyline-vscode/react-ui/src/components/ScatterGraph.js]: Deleted comment

File: skyline-vscode/react-ui/src/data/mock_data.js (hunk 5 above)
Comment (johncalesp) [reply skyline-vscode/react-ui/src/data/mock_data.js]: Sorry, in previous commit I commented for testing purposes, now all is uncommented

File: skyline-vscode/react-ui/src/utils.js (hunk 7 above)
Comment (johncalesp) [reply skyline-vscode/react-ui/src/utils.js]: I used as a guide on how the format should look like, I deleted it now

File: skyline-vscode/react-ui/src/data/providers.js (hunk 9 above)
Comment (johncalesp) [reply skyline-vscode/react-ui/src/data/providers.js]: renamed to gpuPropertyList

File: skyline-vscode/react-ui/src/data/providers.js (hunk 10 above)
Comment (johncalesp) [reply skyline-vscode/react-ui/src/data/providers.js]: renamed to cloudProviders

File: skyline-vscode/react-ui/src/data/providers.js (hunk 11 above)
Comment (johncalesp) [reply skyline-vscode/react-ui/src/data/providers.js]: renamed to cloudInstances

File: skyline-vscode/react-ui/src/App.js [hunk 12]
```diff
@@ -139,14 +139,15 @@ function App() {
 
     useEffect(function () {
         window.addEventListener('message', event => {
-            //console.log("Message:", JSON.stringify(event.data));
             if (event.data['message_type'] === "analysis") {
+                console.log("Received data")
```
Comment (michaelshin) [inline skyline-vscode/react-ui/src/App.js:143]: Don't need to log these messages.

Comment (johncalesp) [reply skyline-vscode/react-ui/src/App.js]: Deleted console log messages in useEffect call

**Review by michaelshin — APPROVED** [review]

**Pull request merged**